### DISCUSSION TITLE
Flow task naming : Parent-child is counter-intuitive

### DISCUSSION THREAD

**Discussion by VaZark** [taskforcesh/bullmq#2073] [General]
Based on the docs

> a parent job will not be moved to the wait status, i.e. can be picked up by a worker, until all its children jobs have been processed successfully. 

I feel like the naming is ambiguous as it can also be taken as "When starting parent task,children tasks are also started". 

Calling the children   `dependsOn` or `beforeTask` would it easier to understand

**Comment by manast** [comment]
I am not sure I understand the ambiguity. A parent job is a job that is placed in the waiting-children set. As soon as all its dependencies (children) are processed, the parent will be moved to waiting status and executed as soon as a worker can pick it up.
